pytorch_lightning/trainer/trainer.py

```diff
@@ -502,11 +502,19 @@ def validate(
 
         model_provided = model is not None
         model = model or self.lightning_module
+        if model is None:
+            raise MisconfigurationException(
+                '`model` must be provided to `trainer.validate()` when `trainer.lightining_module` is not set'
+            )
 
         # links data to the trainer
         self.data_connector.attach_data(model, val_dataloaders=val_dataloaders, datamodule=datamodule)
 
-        if not model_provided:
+        if model_provided:
+            if ckpt_path is not None:
+                raise MisconfigurationException('You cannot pass both `trainer.validate(model=..., ckpt_path=...)`')
```

Comment:
Should we remove this limitation so that if both are passed, we load the ckpt path to the provided model?

Reply:
Yes I believe this is the most natural behavior for the API. Let me try.

Reply:
@carmocca it seems that a bit of refactoring is required to implement the behavior cleanly. I'll open an issue and defer it in another PR.

Reply:
Sounds good to me! Tag me when you do